.github/workflows/check-types.yml

```diff
@@ -0,0 +1,43 @@
+name: Check Types 
+
+on: push
```

Comment:
I think we typically run on both pull-request & pushes to main.

Reply:
so this will run on push to any branch. This is how the ruff check is configured and I figured it makes sense for type checking to have the same policy

Reply:
gotcha.. it'd be good to standardize.. the rest of our stuff all run on PRs.. so it's possible to have a private branch without triggering all the CI. and I think it'll be possible to require checks to pass before allowing merges.. I think we need that.

Reply:
ok.. I think we can already do the latter.. that's a configuration.. I'll enable.

Reply:
cool, I changed it to run on push to main or dev and pr onto main or dev